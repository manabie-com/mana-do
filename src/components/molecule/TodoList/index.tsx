import ToDoList from "./TodoList";

export default ToDoList;
