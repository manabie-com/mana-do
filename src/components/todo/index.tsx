import React, { useEffect, useReducer, useRef, useState, useMemo, useCallback } from 'react';
import { RouteComponentProps } from 'react-router-dom';

import reducer, { initialState } from 'root/store/reducer';
import {
  setTodos,
  createTodo,
  deleteTodo,
  toggleAllTodos,
  deleteAllTodos,
  updateTodoStatus,
  updateTodoContent
} from 'root/store/actions';
import Service from 'root/service';
import { TodoStatus } from 'root/models/todo';
import { isTodoCompleted } from 'root/utils';
import TodoPresentation from './presentation';

import './_todo.css'

type EnhanceTodoStatus = TodoStatus | 'ALL';


const TodoComponent = ({ history }: RouteComponentProps) => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState);
  const [showing, setShowing] = useState<EnhanceTodoStatus>('ALL');
  const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    (async () => {
      const resp = await Service.getTodos();
      dispatch(setTodos(resp || []));
    })()
  }, [])

  useEffect(() => {
    if (todos.length > 0) {
      const lastItem = document.getElementById(`todo-item-${todos[todos.length - 1].id}`)
      if (lastItem) setTimeout(() => {
        lastItem.scrollIntoView({ block: 'end',  behavior: 'smooth' })
      }, 401) // delay to wait for css animation work first
    }
  }, [todos, showing])

  const onCreateTodo = useCallback(async (e: React.KeyboardEvent<HTMLInputElement>) => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    if (e.key === 'Enter' && inputRef.current) {
      try {
        const resp = await Service.createTodo(inputRef.current.value);
        dispatch(createTodo(resp));
        inputRef.current.value = '';
      } catch (e) {
        if (e.response.status === 401) {
          history.push('/')
        }
      }
    }
  }, [dispatch, history])

  const onUpdateTodoStatus: Function = useCallback(async (e: React.ChangeEvent<HTMLInputElement>, todoId: string) => {
    try {
      await Service.updateTodoStatus(todoId, e.target.checked);
      dispatch(updateTodoStatus(todoId, e.target.checked));
    } catch (error) {

    }
  }, [dispatch])

  const onToggleAllTodo = useCallback(async (e: React.ChangeEvent<HTMLInputElement>) => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    try {
      await Service.toggleAllTodos(e.target.checked)
      dispatch(toggleAllTodos(e.target.checked))
    } catch (error) {

    }
  }, [dispatch])

  const onDeleteAllTodo = useCallback(async () => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    try {
      await Service.deleteAllTodo()
      dispatch(deleteAllTodos());
    } catch (error) {

    }
  }, [dispatch])

  const handleDeleteTodo = useCallback(async (todoId: string) => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    try {
      await Service.deleteTodo(todoId)
      dispatch(deleteTodo(todoId));
    } catch (error) {

    }
  }, [dispatch])

  const handleUpdateTodoContent = useCallback(async (todoId: string, content: string) => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    try {
      await Service.updateTodoContent(todoId, content)
      dispatch(updateTodoContent(todoId, content));
    } catch (error) {

    }
  }, [dispatch])

  const handleShowingAll = useCallback(() => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    setShowing('ALL')
  }, [setShowing])

  const handleShowingActive = useCallback(() => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    setShowing(TodoStatus.ACTIVE)
  }, [setShowing])

  const handleShowingCompleted = useCallback(() => {
    /**
     * reduce rebinding function. Does not rebind when dependencies has not changed
     */
    setShowing(TodoStatus.COMPLETED)
  }, [setShowing])

  const showTodos = useMemo(() => todos.filter((todo) => {
    /**
     * Reduce recalculating showing todos, Does not recalculate showing todos when todos and showing state have not changed
     */
    switch (showing) {
      case TodoStatus.ACTIVE:
        return todo.status === TodoStatus.ACTIVE;
      case TodoStatus.COMPLETED:
        return todo.status === TodoStatus.COMPLETED;
      case 'ALL':
      default:
        return true;
    }
  }), [showing, todos])

  const activeTodos = useMemo(() => todos.reduce(function (accum, todo) {
    /**
     * Reduce recalculating activeTodos, Does not recalculate active todos when todos has not changed
    */
    return isTodoCompleted(todo) ? accum : accum + 1;
  }, 0), [todos])

  return (
    <TodoPresentation
      inputRef={inputRef}
      onCreateTodo={onCreateTodo}
      showTodos={showTodos}
      onUpdateTodoStatus={onUpdateTodoStatus}
      deleteTodo={handleDeleteTodo}
      todos={todos}
      activeTodos={activeTodos}
      onToggleAllTodo={onToggleAllTodo}
      handleShowingAll={handleShowingAll}
      handleShowingActive={handleShowingActive}
      handleShowingCompleted={handleShowingCompleted}
      onDeleteAllTodo={onDeleteAllTodo}
      onUpdateTodoContent={handleUpdateTodoContent}
      showing={showing}
    />
  );
};

export default TodoComponent;