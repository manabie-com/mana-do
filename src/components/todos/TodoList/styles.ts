import styled from "styled-components";

export const LoaderWrap = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  overflow: hidden;

  .loader {
    font-size: 10px;
    margin-top: 32px;
    margin-bottom: 50px;
    text-indent: -9999em;
    width: 5em;
    height: 5em;
    border-radius: 50%;
    background: var(--brightblue);
    background: -moz-linear-gradient(left, var(--brightblue) 10%, rgba(128, 0, 255, 0) 42%);
    background: -webkit-linear-gradient(left, var(--brightblue) 10%, rgba(128, 0, 255, 0) 42%);
    background: -o-linear-gradient(left, var(--brightblue) 10%, rgba(128, 0, 255, 0) 42%);
    background: -ms-linear-gradient(left, var(--brightblue) 10%, rgba(128, 0, 255, 0) 42%);
    background: linear-gradient(to right, var(--brightblue) 10%, rgba(128, 0, 255, 0) 42%);
    position: relative;
    -webkit-animation: load3 1.4s infinite linear;
    animation: load3 1.4s infinite linear;
    -webkit-transform: translateZ(0);
    -ms-transform: translateZ(0);
    transform: translateZ(0);

    :before {
      width: 50%;
      height: 50%;
      background: var(--brightblue);
      border-radius: 100% 0 0 0;
      position: absolute;
      top: 0;
      left: 0;
      content: "";
    }

    :after {
      background: #ffffff;
      width: 75%;
      height: 75%;
      border-radius: 50%;
      content: "";
      margin: auto;
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
    }
  }

  @-webkit-keyframes load3 {
    0% {
      -webkit-transform: rotate(0deg);
      transform: rotate(0deg);
    }
    100% {
      -webkit-transform: rotate(360deg);
      transform: rotate(360deg);
    }
  }

  @keyframes load3 {
    0% {
      -webkit-transform: rotate(0deg);
      transform: rotate(0deg);
    }
    100% {
      -webkit-transform: rotate(360deg);
      transform: rotate(360deg);
    }
  }
`;

export const Wrapper = styled.div`
  margin: 0 auto;
  width: var(--todo-width);

  @media screen and (max-width: 768px) {
    max-height: calc(100vh - 370px);
    overflow-y: auto;
  }
`;
