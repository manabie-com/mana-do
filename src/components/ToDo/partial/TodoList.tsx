import React, { useState, useEffect } from 'react';

import { TodoStatus, EditTodo, Todo } from '../../../models/todo';
import { getRandomInt } from '../../../helpers';

import {
    updateTodo,
    updateTodoStatus,
    deleteTodo
} from '../../../store/actions';

import { MOTIV_MSG1, MOTIV_MSG2, MOTIV_MSG3, COMPLETED_TASK_MSG, NOT_YET_DONE_MSG } from '../../../constants';

const ALL: string = "ALL";
type EnhanceTodoStatus = TodoStatus | typeof ALL;

interface TodoListTypes {
    todos: Array<Todo>,
    showing: EnhanceTodoStatus,
    setShowing: any,
    dispatch: any
};

const motivationalMsg: string[] = [
    MOTIV_MSG1,
    MOTIV_MSG2,
    MOTIV_MSG3,
];

const TodoList = ({ todos, showing, setShowing, dispatch }: TodoListTypes) => {
    const [motivationMessage, setMotivationalMessage] = useState<string>(motivationalMsg[0]);
    const [editTodo, setEditTodo] = useState<EditTodo>({ editMode: false, id: null, content: null });

    useEffect(() => {
        if (showing === TodoStatus.COMPLETED) {
            setMotivationalMessage(motivationalMsg[getRandomInt(3)]);
        }
    }, [showing]);

    const onDeleteTodoById = (todiId: any) => dispatch(deleteTodo(todiId));

    const onChangeEditTodo = (id: any, content: any) => setEditTodo(prev => ({ ...prev, id, content }));

    const handleEditTodo = (id: any, content: any) => setEditTodo({ editMode: true, id, content });

    const onUpdateTodo = (id: any, content: any) => {
        let data = { editMode: false, id, content };
        setEditTodo(prev => ({ ...prev, ...data }));
        dispatch(updateTodo(data));
    };

    const onUpdateTodoStatus = (e: React.ChangeEvent<HTMLInputElement>, todoId: any) => dispatch(updateTodoStatus(todoId, e.target.checked));

    const filterTodosList = (status: string) => {
        if (showing === ALL) {
            return true
        } else if (showing === status) {
            return true;
        } else {
            return false;
        }
    };

    const checkTodosByStatus = () => {
        const result = todos.find(item => item.status === showing);

        if (!result && showing !== ALL) {
            if (todos.length > 0) {
                if (showing === TodoStatus.COMPLETED) {
                    return <>
                        <small>{NOT_YET_DONE_MSG}</small>
                        <p>{motivationMessage}</p>
                    </>;
                } else {
                    return COMPLETED_TASK_MSG;
                }
            }
        }
    };

    return (
        <div className="ToDo__list">
            {   
                todos.map((todo, index) =>
                    filterTodosList(todo.status) &&
                    <div
                        key={index}
                        className="ToDo__item"
                        data-testid="todo__item"
                    >
                        <input
                            type="checkbox"
                            checked={todo.status === TodoStatus.COMPLETED}
                            onChange={e => onUpdateTodoStatus(e, todo.id)}
                        />
                        <span className={`todo__${todo.status}`} onDoubleClick={() => handleEditTodo(todo.id, todo.content)}>
                            {
                                (editTodo.editMode && editTodo.id === todo.id)
                                ? <input
                                    value={editTodo.content}
                                    onChange={e => onChangeEditTodo(todo.id, e.target.value)}
                                    onBlur={() => onUpdateTodo(todo.id, editTodo.content)}
                                    style={{ width: '95%' }}
                                />
                                : todo.content
                            }
                        </span>
                        <button
                            className="Todo__delete"
                            onClick={() => onDeleteTodoById(todo.id)}
                        >
                            X
                        </button>
                    </div>
                )
            }
            {checkTodosByStatus()}
            {
                todos.length === 0
                && <div
                        className='toDo__message'
                        data-testid='no__todo__message'
                    >
                        <h4>MANABIE TO DO</h4>
                        <p>Free up your mental space, gives you focus, from work to play.</p>
                    </div>
            }
        </div>
    )
};

export default TodoList;