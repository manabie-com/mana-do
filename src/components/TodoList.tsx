import React, {Dispatch, SetStateAction} from "react";
import {AppActions,} from "store/actions";
import {Todo, TodoStatus} from "models/todo";
import TodoItem from "./TodoItem";

type EnhanceTodoStatus = TodoStatus | 'ALL';

interface ToolBarProps {
    dispatch: Dispatch<AppActions>,
    todos: Array<Todo>,
    showing: EnhanceTodoStatus,
    setShowing: Dispatch<SetStateAction<EnhanceTodoStatus>>
}

const TodoList: React.FC<ToolBarProps> = ({dispatch, todos, showing, setShowing}) => {
    const todosFilter = todos.filter((todo) => {
        switch (showing) {
            case TodoStatus.ACTIVE:
                return todo.status === TodoStatus.ACTIVE;
            case TodoStatus.COMPLETED:
                return todo.status === TodoStatus.COMPLETED;
            default:
                return true;
        }
    });

    return (
        <div className="Todo__list">
            {todosFilter.map((todo, index) => {
                const todoItemProps = {dispatch, todo}
                return <TodoItem key={todo.id} {...todoItemProps} />;
            })}
            {todos.length > 0 && !(todosFilter.length > 0) &&
            <div className="__no-todos">{"no " + (showing === TodoStatus.ACTIVE ? 'active' : 'completed') + " todos..."}</div>}
        </div>
    );
};

export default TodoList;
