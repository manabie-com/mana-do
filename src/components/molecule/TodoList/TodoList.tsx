import { Todo, TodoStatus } from "models/todo";
import React from "react";
import ToDoItem from "components/atom/TodoItem";

export type EnhanceTodoStatus = TodoStatus | "ALL";

type ToDoListProps = {
  showTodos: Todo[];
  showing: EnhanceTodoStatus;
};

export const emptyTodoMessage = (showing: EnhanceTodoStatus) =>
  `There are not${
    showing === TodoStatus.ACTIVE
      ? " active"
      : showing === TodoStatus.COMPLETED
      ? " completed"
      : ""
  } items`;

const ToDoList: React.FC<ToDoListProps> = ({ showTodos, showing }) => {
  return (
    <div className="ToDo__list">
      {showTodos.length === 0 && (
        <span className="Todo__empty__message" data-testid="empty-message">
          {emptyTodoMessage(showing)}
        </span>
      )}
      {showTodos.map((todo) => {
        return <ToDoItem todo={todo} key={todo.id} />;
      })}
    </div>
  );
};

export default ToDoList;
