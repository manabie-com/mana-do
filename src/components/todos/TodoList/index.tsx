import React from "react";

import { TodoLoadingStatus, TodoStatus } from "../../../models/todo";
import { useAppContext } from "../../../AppContext";
import { DragDropContext, Droppable } from "react-beautiful-dnd";
import TodoListItems from "../TodoListItems";
import { LoaderWrap, Wrapper } from "./styles";
import { setTodos } from "../../../store/actions";
import { TODO_ITEM_HEIGHT } from "../../../utils/constants";

const TodoList = () => {
  const {
    state: {
      todos,
      todosLoadStatus,
      filter: { status },
    },
    dispatch,
  } = useAppContext();

  if (todosLoadStatus === TodoLoadingStatus.Loading) {
    return (
      <LoaderWrap>
        <div className="loader">Loading...</div>
      </LoaderWrap>
    );
  }

  const showTodos = todos.filter((todo) => {
    switch (status) {
      case TodoStatus.Active:
        return todo.status === TodoStatus.Active;
      case TodoStatus.Completed:
        return todo.status === TodoStatus.Completed;
      default:
        return true;
    }
  });
  const renderedListItems = showTodos.map((todo, index) => <TodoListItems key={todo.id} id={todo.id} index={index} />);

  return (
    <Wrapper style={{ height: showTodos.length * TODO_ITEM_HEIGHT }}>
      <DragDropContext
        onDragEnd={(data) => {
          const srcIndex = data.source.index;
          const destIndex = data.destination?.index || 0;
          console.log({ srcIndex, destIndex });

          // sort
          const temp = [...todos];
          console.log(temp);
          temp.splice(destIndex, 0, temp.splice(srcIndex, 1)[0]);
          dispatch(setTodos(temp));
        }}
      >
        <Droppable droppableId="droppable-1">
          {(provided, _) => (
            <div ref={provided.innerRef} {...provided.droppableProps}>
              {renderedListItems}
            </div>
          )}
        </Droppable>
      </DragDropContext>
    </Wrapper>
  );
};

export default TodoList;
