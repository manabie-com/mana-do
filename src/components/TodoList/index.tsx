import React, { useContext } from "react";
import { AppContext } from "../../store/context";
import TodoItem from './TodoItem';
import './TodoList.css';
interface ITodoListProps {
    filter:string,
    [x:string]:string | any
}

const TodoList = ({filter}:ITodoListProps) => {
    const { state } = useContext(AppContext);
    const { todos } = state; 
    const filteredTodo = todos.filter(item => filter!=='ALL'?item.status === filter:item);
    return (<div className="table-container"><table className="app-table">
            <thead>
                <tr>
                    <th className="th-seq-number align-center">No</th>
                    <th className="th-seq-number align-center">Done?</th>
                    <th className="th-300px align-left">Content</th>
                    <th className="th-100px align-center">Status</th>
                    <th className="th-100px align-center">Created</th>
                    <th className="th-160px align-center">Action</th>
                </tr>
            </thead>
            <tbody>
                    { filteredTodo.map((item,index)=>{
                        return <TodoItem itemNo={index+1} key={item.id} todo={item}  />
                    }
                    ) }
            </tbody>
        </table></div>)
}

export default TodoList;


