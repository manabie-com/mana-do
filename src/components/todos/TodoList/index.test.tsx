import React from "react";
import renderer from "react-test-renderer";

import TodoList from ".";
import * as AppContext from "../../../AppContext";
import { TodoLoadingStatus, TodoStatus } from "../../../models/todo";

jest.mock("../TodoListItems", () => "TodoListItems");

jest.mock("react-beautiful-dnd", () => ({
  Droppable: ({ children }: any) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {}
    ),
  Draggable: ({ children }: any) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {}
    ),
  DragDropContext: ({ children }: any) => children,
}));

const mockDispatch = jest.fn();

const mockedTodos = [
  {
    id: "hfuwyr",
    content: "todo 1",
    user_id: "firstUser",
    status: TodoStatus.All,
    created_date: "",
  },
  {
    id: "orislaj",
    content: "todo 2",
    user_id: "firstUser",
    status: TodoStatus.All,
    created_date: "",
  },
  {
    id: "jiwuro",
    content: "todo 3",
    user_id: "firstUser",
    status: TodoStatus.Active,
    created_date: "",
  },
  {
    id: "iejals",
    content: "todo 4",
    user_id: "firstUser",
    status: TodoStatus.Active,
    created_date: "",
  },
  {
    id: "jfuqos",
    content: "todo 5",
    user_id: "firstUser",
    status: TodoStatus.Completed,
    created_date: "",
  },
  {
    id: "ahsyel",
    content: "todo 6",
    user_id: "firstUser",
    status: TodoStatus.Completed,
    created_date: "",
  },
];

describe("snapshot", () => {
  const mockContextValue = {
    state: {
      todos: mockedTodos,
      filter: {},
    },
    dispatch: mockDispatch,
  };

  test("should return snapshot when fetched data done", () => {
    const newMockContextValue = {
      ...mockContextValue,
      state: {
        ...mockContextValue.state,
        todosLoadStatus: TodoLoadingStatus.Idle,
      },
    };
    jest
      .spyOn(AppContext, "useAppContext")
      // @ts-ignore
      .mockImplementation(() => newMockContextValue);

    const tree = renderer.create(<TodoList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should return snapshot when fetching data not done yet", () => {
    const newMockContextValue = {
      ...mockContextValue,
      state: {
        ...mockContextValue.state,
        todosLoadStatus: TodoLoadingStatus.Loading,
      },
    };
    jest
      .spyOn(AppContext, "useAppContext")
      // @ts-ignore
      .mockImplementation(() => newMockContextValue);
    const tree = renderer.create(<TodoList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("todos filter by status", () => {
  // test("should return correct number of todo item with status is ALL", () => {
  //   const mockContextValue = {
  //     state: {
  //       theme: Themes.Light,
  //       todos: mockedTodos,
  //       todosLoadStatus: TodoLoadingStatus.Idle,
  //       filter: {
  //         status: TodoStatus.All,
  //       },
  //     },
  //     dispatch: mockDispatch,
  //   };
  //   jest.spyOn(AppContext, "useAppContext").mockImplementation(() => mockContextValue);
  //   const wrapper = shallow(<TodoList />);
  //   const TodoListItems = wrapper.find("TodoListItems");
  //   expect(TodoListItems.length).toBe(mockedTodos.length);
  // });
  // test("should return correct number of todo item with status is ACTIVE", () => {
  //   const mockContextValue = {
  //     state: {
  //       todos: mockedTodos,
  //       todosLoadStatus: TodoLoadingStatus.Idle,
  //       filter: {
  //         status: TodoStatus.Active,
  //       },
  //     },
  //     dispatch: mockDispatch,
  //   };
  //   jest
  //     .spyOn(AppContext, "useAppContext")
  //     // @ts-ignore
  //     .mockImplementation(() => mockContextValue);
  //   const wrapper = shallow(<TodoList />);
  //   const TodoListItems = wrapper.find("Draggable");
  //   expect(TodoListItems.length).toBe(2);
  // });
  // test("should return correct number of todo item with status is COMPLETED", () => {
  //   const mockContextValue = {
  //     state: {
  //       todos: mockedTodos,
  //       todosLoadStatus: TodoLoadingStatus.Idle,
  //       filter: {
  //         status: TodoStatus.Completed,
  //       },
  //     },
  //     dispatch: mockDispatch,
  //   };
  //   jest
  //     .spyOn(AppContext, "useAppContext")
  //     // @ts-ignore
  //     .mockImplementation(() => mockContextValue);
  //   const wrapper = shallow(<TodoList />);
  //   const TodoListItems = wrapper.find("Draggable");
  //   expect(TodoListItems.length).toBe(2);
  // });
});
