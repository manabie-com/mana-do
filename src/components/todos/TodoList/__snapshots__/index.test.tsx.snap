// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should return snapshot when fetched data done 1`] = `
<div
  className="sc-gsnTZi esZDVH"
  style={
    Object {
      "height": 348,
    }
  }
>
  <div>
    <TodoListItems
      id="hfuwyr"
      index={0}
    />
    <TodoListItems
      id="orislaj"
      index={1}
    />
    <TodoListItems
      id="jiwuro"
      index={2}
    />
    <TodoListItems
      id="iejals"
      index={3}
    />
    <TodoListItems
      id="jfuqos"
      index={4}
    />
    <TodoListItems
      id="ahsyel"
      index={5}
    />
  </div>
</div>
`;

exports[`snapshot should return snapshot when fetching data not done yet 1`] = `
<div
  className="sc-bczRLJ ibKHch"
>
  <div
    className="loader"
  >
    Loading...
  </div>
</div>
`;
